### PR TITLE
Date time token in Toml source

### LINKED ISSUE #15364: Add date / time tokens in Toml Module
This came up in developement of #15228. 
Inline comments in a TOML module were treated as newline/ fullline comments due to which tests of date / time which had inline comments got passed. eg : 
```
 timestamp3 = 06:30:30.1234567 # rounding up
```
was treated as 
```
 timestamp3 = 06:30:30.1234567 
# rounding up
```
and the token for the timestamp would be ` (06:30:30.1234567)`. 
But with the merge of #15228 , 
inline comment would not be considered as fullline and the token would be `(06:30:30.1234567 # rounding up)` which is an invalid token. This occurs because it has not been specified to separate date or time tokens in a line.

### PR DESCRIPTION
**Pull request by ankingcodes**
Added a check in class `Source` of TOML module to create separate tokens for date/time. 
Earlier, for a line `timestamp3 = 06:30:30.1234567 # rounding up` the token created was `(06:30:30.1234567 # rounding up)` which is an invalid token, after merge of #15228 .
Now, token created would be `(06:30:30.1234567)` which is a valid token.

fixes #15364

### CODE REVIEW

**Comment by ankingcodes** [comment]
@ben-albrecht The test results are as follows: 
- test ( fixed the testtime.chpl test error in #15228 )
```
[Test Summary - 200330.213552] 
[Summary: #Successes = 20 | #Failures = 0 | #Futures = 0 | #Warnings = 0 ]
[Summary: #Passing Suppressions = 0 | #Passing Futures = 0 ]
[END]

```
- Burnt Sushi
```
[Test Summary - 200330.213830] 
[Summary: #Successes = 7 | #Failures = 0 | #Futures = 0 | #Warnings = 0 ]
[Summary: #Passing Suppressions = 0 | #Passing Futures = 0 ]
[END]
```
Please give this a review and test before merge or approval. 
This may be merged post-release. 👍

File: test/library/packages/TOML/test/commentTest.compopts [hunk 1]
```diff
@@ -0,0 +1 @@
+-M ../../../modules/packages/
```
Comment (Rapiz1) [inline test/library/packages/TOML/test/commentTest.compopts]: Oops. Missing a blank line here.

Comment (ben-albrecht) [reply test/library/packages/TOML/test/commentTest.compopts]: Thanks @Rapiz1 - This shouldn't impact the compopts format, but would impact a `.good`.

File: modules/packages/TOML.chpl [hunk 2]
```diff
@@ -165,8 +165,8 @@ module TomlParser {
         if !source.isEmpty() {
           while(readLine(source)) {
             var token = top(source);
-
-            if token == '#' {
+
```
Comment (ben-albrecht) [inline modules/packages/TOML.chpl]: nit: remove trailing white space

File: modules/packages/TOML.chpl [hunk 3]
```diff
@@ -1205,26 +1206,42 @@ module TomlReader {
...
         var strippedToken = token.strip(" \t");
         if strippedToken.size != 0 {
           if debugTomlReader {
             writeln('Tokenized: ', '(', strippedToken, ')');
           }
-
           nonEmptyChar = true;
-          linetokens.append(strippedToken);
+          // check for date/time in a line and avoid comment        
+          var toke = strippedToken;
```
Comment (ben-albrecht) [inline modules/packages/TOML.chpl]: nit: If a variable is not intended to change, you can make it a `const` for safety.

File: modules/packages/TOML.chpl (hunk 2 above)
Comment (ben-albrecht) [reply modules/packages/TOML.chpl]: pro-tip: configure your editor to display trailing white space to avoid this pitfall in the future.

**Comment by ankingcodes** [comment]
@Rapiz1 @ben-albrecht thanks for your comments. I'll make the suggested changes. When is the new  release scheduled ?

**Comment by Rapiz1** [comment]
See https://chapel-lang.org/events.html for the release schedule.

**Comment by ben-albrecht** [comment]
>> When is the new release scheduled ?

> See https://chapel-lang.org/events.html for the release schedule.

Thanks @Rapiz1. Also note, we are doing a special follow-up 1.22 release that makes the switch to 0-based indices, so the code freeze period will last a bit longer than normal for this release. There will be some announcements about this coming soon.

**Comment by ben-albrecht** [comment]
@ankingcodes - Can you rebase with master and confirm this PR works with 0-based indices?

**Comment by ankingcodes** [comment]
@ben-albrecht closing this branch here because of conflicts with #15228, change done at 72752da

**Pull request closed without merging**